<?xml version="1.0" encoding="UTF-8"?>
<sdf version='1.10'>
  <world name='warehouse_outdoor'>
    <gravity>0 0 -9.8000000000000007</gravity>
    <physics type='ode'>
      <max_step_size>0.004</max_step_size>
      <real_time_factor>1.0</real_time_factor>
      <real_time_update_rate>250</real_time_update_rate>
    </physics>
    
    <gui fullscreen='false'>
      <plugin name='3D View' filename='GzScene3D'>
        <ignition-gui>
          <title>3D View</title>
          <property type='bool' key='showTitleBar'>false</property>
          <property type='string' key='state'>docked</property>
        </ignition-gui>
        <engine>ogre2</engine>
        <scene>scene</scene>
        <ambient_light>0.4 0.4 0.4</ambient_light>
        <background_color>0.8 0.8 0.8</background_color>
        <camera_pose>13.4 -6.1 15 0 0.6 -1.83</camera_pose>
      </plugin>
      <plugin name='World control' filename='WorldControl'>
        <ignition-gui>
          <title>World control</title>
          <property type='bool' key='showTitleBar'>false</property>
          <property type='bool' key='resizable'>false</property>
          <property type='double' key='height'>72</property>
          <property type='double' key='width'>121</property>
          <property type='double' key='z'>1</property>
          <property type='string' key='state'>floating</property>
          <anchors target='3D View'>
            <line own='left' target='left'/>
            <line own='bottom' target='bottom'/>
          </anchors>
        </ignition-gui>
        <play_pause>true</play_pause>
        <step>true</step>
        <start_paused>true</start_paused>
      </plugin>
      <plugin name='World stats' filename='WorldStats'>
        <ignition-gui>
          <title>World stats</title>
          <property type='bool' key='showTitleBar'>false</property>
          <property type='bool' key='resizable'>false</property>
          <property type='double' key='height'>110</property>
          <property type='double' key='width'>290</property>
          <property type='double' key='z'>1</property>
          <property type='string' key='state'>floating</property>
          <anchors target='3D View'>
            <line own='right' target='right'/>
            <line own='bottom' target='bottom'/>
          </anchors>
        </ignition-gui>
        <sim_time>true</sim_time>
        <real_time>true</real_time>
        <real_time_factor>true</real_time_factor>
        <iterations>true</iterations>
      </plugin>
      <plugin name='Transform control' filename='TransformControl'>
        <ignition-gui>
          <property type='bool' key='showTitleBar'>false</property>
          <property key='x' type='double'>0</property>
          <property key='y' type='double'>0</property>
          <property key='width' type='double'>250</property>
          <property key='height' type='double'>50</property>
          <property key='state' type='string'>floating</property>
          <property key='cardBackground' type='string'>#666666</property>
        </ignition-gui>
      </plugin>
      <plugin name='Shapes' filename='Shapes'>
        <ignition-gui>
          <property type='bool' key='showTitleBar'>false</property>
          <property key='x' type='double'>250</property>
          <property key='y' type='double'>0</property>
          <property key='width' type='double'>150</property>
          <property key='height' type='double'>50</property>
          <property key='state' type='string'>floating</property>
          <property key='cardBackground' type='string'>#666666</property>
        </ignition-gui>
      </plugin>
      <plugin name='Screenshot' filename='Screenshot'>
        <ignition-gui>
          <property type='bool' key='showTitleBar'>false</property>
          <property key='x' type='double'>400</property>
          <property key='y' type='double'>0</property>
          <property key='width' type='double'>50</property>
          <property key='height' type='double'>50</property>
          <property key='state' type='string'>floating</property>
          <property key='cardBackground' type='string'>#666666</property>
        </ignition-gui>
      </plugin>
      <plugin name='Component inspector' filename='ComponentInspector'>
        <ignition-gui>
          <property key='x' type='double'>0</property>
          <property key='y' type='double'>0</property>
          <property key='width' type='double'>400</property>
          <property key='height' type='double'>375</property>
          <property key='state' type='string'>docked_collapsed</property>
        </ignition-gui>
      </plugin>
      <plugin name='Entity tree' filename='EntityTree'>
        <ignition-gui>
          <property key='x' type='double'>0</property>
          <property key='y' type='double'>0</property>
          <property key='width' type='double'>400</property>
          <property key='height' type='double'>375</property>
          <property key='state' type='string'>docked_collapsed</property>
        </ignition-gui>
      </plugin>
    </gui>
    <scene>
      <ambient>0.8 0.8 0.8 1</ambient>
      <background>0.6 0.8 1.0 1</background>
      <shadows>true</shadows>
      <grid>false</grid>
    </scene>
    <magnetic_field>6e-06 2.3e-05 -4.2e-05</magnetic_field>
    <atmosphere type='adiabatic'/>

    <spherical_coordinates>
      <surface_model>EARTH_WGS84</surface_model>
      <world_frame_orientation>ENU</world_frame_orientation>
      <latitude_deg>10.772175109674038</latitude_deg>
      <longitude_deg>106.65789107082472</longitude_deg>
      <elevation>0</elevation>
    </spherical_coordinates>
    
    <!-- Large outdoor ground plane -->
    <model name='ground_plane'>
      <static>true</static>
      <link name='link'>
        <collision name='collision'>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>200 200</size>
            </plane>
          </geometry>
          <surface>
            <friction>
              <ode/>
            </friction>
            <bounce/>
            <contact/>
          </surface>
        </collision>
        <visual name='visual'>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>200 200</size>
            </plane>
          </geometry>
          <material>
            <ambient>0.5 0.7 0.3 1</ambient>
            <diffuse>0.5 0.7 0.3 1</diffuse>
            <specular>0 0 0 1</specular>
          </material>
        </visual>
        <pose>0 0 0 0 0 0</pose>
        <inertial>
          <pose>0 0 0 0 0 0</pose>
          <mass>1</mass>
          <inertia>
            <ixx>1</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>1</iyy>
            <iyz>0</iyz>
            <izz>1</izz>
          </inertia>
        </inertial>
        <enable_wind>false</enable_wind>
      </link>
      <pose>0 0 0 0 0 0</pose>
      <self_collide>false</self_collide>
    </model>
    
    <!-- Warehouse building without roof for GPS access -->
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Warehouse</uri>
      <name>warehouse</name>
      <pose>0 0 -0.089999999999999997 3.14159 3.14159 3.14159</pose>
    </include>
    
    <!-- Warehouse contents -->
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/cart_model_2</uri>
      <name>cart1</name>
      <pose>-5.7300000000000004 15 0.25 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_0</name>
      <pose>-4.4152800000000001 2.3069700000000002 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_1</name>
      <pose>-4.4152800000000001 5.30708 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_2</name>
      <pose>-4.4152800000000001 8.3435199999999998 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_3</name>
      <pose>5.6014400000000002 8.3435199999999998 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_4</name>
      <pose>5.6014400000000002 5.30708 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_5</name>
      <pose>5.6014400000000002 -0.69098700000000002 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_6</name>
      <pose>5.6014400000000002 2.3069700000000002 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/pallet_box_mobile</uri>
      <name>pallet_box_mobile</name>
      <pose>4.4161000000000001 14.6952 0.01 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/pallet_box_mobile</uri>
      <name>pallet_box_mobile_0</name>
      <pose>4.4541500000000003 13.6212 0.01 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/pallet_box_mobile</uri>
      <name>pallet_box_mobile_1</name>
      <pose>4.4467999999999996 12.228999999999999 0.01 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/pallet_box_mobile</uri>
      <name>pallet_box</name>
      <pose>-6.1191300000000002 13.7079 0.01 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf_big</uri>
      <name>shelf_big_1</name>
      <pose>13.982100000000001 15.319000000000001 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf_big</uri>
      <name>shelf_big_2</name>
      <pose>6.1977700000000002 -12.964700000000001 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf_big</uri>
      <name>shelf_big_3</name>
      <pose>0.59437600000000002 -12.964700000000001 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf_big</uri>
      <name>shelf_big_4</name>
      <pose>-5.3628400000000003 -12.964700000000001 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_7</name>
      <pose>13.3818 -21.241599999999998 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_8</name>
      <pose>13.3818 -19.002800000000001 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_9</name>
      <pose>13.3818 -16.447800000000001 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/shelf</uri>
      <name>shelf_10</name>
      <pose>13.3818 -14.1028 0 0 0 0</pose>
    </include>
    <include>
      <uri>https://fuel.ignitionrobotics.org/1.0/MovAi/models/pallet_box_mobile</uri>
      <name>pallet_box_0</name>
      <pose>14.0222 -24.335000000000001 0.01 0 0 0</pose>
    </include>
    
    <!-- QR codes for shelf_big_4 (pose: -5.363, -12.965, 0) -->
    <model name="qr_shelfbig4_1"><static>true</static><pose>-4.9 -4.5 2.2 0 0 0</pose><link name="qr_link"><visual name="qr_visual"><geometry><plane><normal>1 0 0</normal><size>0.15 0.15</size></plane></geometry><material><ambient>1 1 1 1</ambient><diffuse>1 1 1 1</diffuse><pbr><metal><albedo_map>resources/QR/qr.png</albedo_map></metal></pbr></material></visual></link></model>
    <model name="qr_shelfbig4_2"><static>true</static><pose>-4.9 -5.4 2.2 0 0 0</pose><link name="qr_link"><visual name="qr_visual"><geometry><plane><normal>1 0 0</normal><size>0.15 0.15</size></plane></geometry><material><ambient>1 1 1 1</ambient><diffuse>1 1 1 1</diffuse><pbr><metal><albedo_map>resources/QR/qr.png</albedo_map></metal></pbr></material></visual></link></model>
    <model name="qr_shelfbig4_3"><static>true</static><pose>-4.88 -6.3 2.2 0 0 0</pose><link name="qr_link"><visual name="qr_visual"><geometry><plane><normal>1 0 0</normal><size>0.15 0.15</size></plane></geometry><material><ambient>1 1 1 1</ambient><diffuse>1 1 1 1</diffuse><pbr><metal><albedo_map>resources/QR/qr.png</albedo_map></metal></pbr></material></visual></link></model>
    <model name="qr_shelfbig4_4"><static>true</static><pose>-4.9 -7.2 2.2 0 0 0</pose><link name="qr_link"><visual name="qr_visual"><geometry><plane><normal>1 0 0</normal><size>0.15 0.15</size></plane></geometry><material><ambient>1 1 1 1</ambient><diffuse>1 1 1 1</diffuse><pbr><metal><albedo_map>resources/QR/qr.png</albedo_map></metal></pbr></material></visual></link></model>
    <model name="qr_shelfbig4_5"><static>true</static><pose>-4.9 -8.1 2.2 0 0 0</pose><link name="qr_link"><visual name="qr_visual"><geometry><plane><normal>1 0 0</normal><size>0.15 0.15</size></plane></geometry><material><ambient>1 1 1 1</ambient><diffuse>1 1 1 1</diffuse><pbr><metal><albedo_map>resources/QR/qr.png</albedo_map></metal></pbr></material></visual></link></model>
    <model name="qr_shelfbig4_6"><static>true</static><pose>-4.9 -5.4 2.81 0 0 0</pose><link name="qr_link"><visual name="qr_visual"><geometry><plane><normal>1 0 0</normal><size>0.15 0.15</size></plane></geometry><material><ambient>1 1 1 1</ambient><diffuse>1 1 1 1</diffuse><pbr><metal><albedo_map>resources/QR/qr.png</albedo_map></metal></pbr></material></visual></link></model>
    <model name="qr_shelfbig4_7"><static>true</static><pose>-4.9 -6.7 2.81 0 0 0</pose><link name="qr_link"><visual name="qr_visual"><geometry><plane><normal>1 0 0</normal><size>0.15 0.15</size></plane></geometry><material><ambient>1 1 1 1</ambient><diffuse>1 1 1 1</diffuse><pbr><metal><albedo_map>resources/QR/qr.png</albedo_map></metal></pbr></material></visual></link></model>
    <model name="qr_shelfbig4_8"><static>true</static><pose>-4.9 -7.9 2.81 0 0 0</pose><link name="qr_link"><visual name="qr_visual"><geometry><plane><normal>1 0 0</normal><size>0.15 0.15</size></plane></geometry><material><ambient>1 1 1 1</ambient><diffuse>1 1 1 1</diffuse><pbr><metal><albedo_map>resources/QR/qr.png</albedo_map></metal></pbr></material></visual></link></model>
    
    <!-- Outdoor lighting - sun positioned high for GPS simulation -->
    <light name='sun' type='directional'>
      <pose>0 0 50 0 0 0</pose>
      <cast_shadows>true</cast_shadows>
      <intensity>1.2</intensity>
      <direction>0 0 -1</direction>
      <diffuse>1 1 1 1</diffuse>
      <specular>0.8 0.8 0.8 1</specular>
      <attenuation>
        <range>1000</range>
        <linear>0</linear>
        <constant>1</constant>
        <quadratic>0</quadratic>
      </attenuation>
    </light>
    
    <!-- Additional outdoor elements -->
    <model name='tree_1'>
      <static>true</static>
      <pose>-30 -30 0 0 0 0</pose>
      <link name='link'>
        <visual name='trunk'>
          <geometry>
            <cylinder>
              <radius>0.5</radius>
              <length>8</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.4 0.2 0.1 1</ambient>
            <diffuse>0.4 0.2 0.1 1</diffuse>
          </material>
          <pose>0 0 4 0 0 0</pose>
        </visual>
        <visual name='leaves'>
          <geometry>
            <sphere>
              <radius>4</radius>
            </sphere>
          </geometry>
          <material>
            <ambient>0.1 0.6 0.1 1</ambient>
            <diffuse>0.1 0.6 0.1 1</diffuse>
          </material>
          <pose>0 0 10 0 0 0</pose>
        </visual>
        <collision name='trunk_collision'>
          <geometry>
            <cylinder>
              <radius>0.5</radius>
              <length>8</length>
            </cylinder>
          </geometry>
          <pose>0 0 4 0 0 0</pose>
        </collision>
      </link>
    </model>
    
    <model name='tree_2'>
      <static>true</static>
      <pose>30 30 0 0 0 0</pose>
      <link name='link'>
        <visual name='trunk'>
          <geometry>
            <cylinder>
              <radius>0.5</radius>
              <length>8</length>
            </cylinder>
          </geometry>
          <material>
            <ambient>0.4 0.2 0.1 1</ambient>
            <diffuse>0.4 0.2 0.1 1</diffuse>
          </material>
          <pose>0 0 4 0 0 0</pose>
        </visual>
        <visual name='leaves'>
          <geometry>
            <sphere>
              <radius>4</radius>
            </sphere>
          </geometry>
          <material>
            <ambient>0.1 0.6 0.1 1</ambient>
            <diffuse>0.1 0.6 0.1 1</diffuse>
          </material>
          <pose>0 0 10 0 0 0</pose>
        </visual>
        <collision name='trunk_collision'>
          <geometry>
            <cylinder>
              <radius>0.5</radius>
              <length>8</length>
            </cylinder>
          </geometry>
          <pose>0 0 4 0 0 0</pose>
        </collision>
      </link>
    </model>
    
    <!-- Origin marker -->
    <model name="origin_marker">
      <static>true</static>
      <pose>0 0 0 0 0 0</pose>
      <link name="link">
        <visual name="marker">
          <geometry>
            <sphere>
              <radius>0.05</radius>
            </sphere>
          </geometry>
          <material>
            <ambient>1 0 1 1</ambient>
            <diffuse>1 0 1 1</diffuse>
          </material>
        </visual>

        <visual name="x_axis">
          <geometry>
            <cylinder>
              <radius>0.02</radius>
              <length>24</length>
            </cylinder>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Red</name>
            </script>
          </material>
          <pose>12 0 0 0 -1.5708 0</pose>
        </visual>
        <visual name="y_axis">
          <geometry><cylinder><radius>0.02</radius><length>24</length></cylinder></geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Green</name>
            </script>
          </material>
          <pose>0 12 0 -1.5708 0 0</pose>
        </visual>
        <visual name="z_axis">
          <geometry><cylinder><radius>0.02</radius><length>24</length></cylinder></geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Blue</name>
            </script>
          </material>
          <pose>0 0 12 0 0 0</pose>
        </visual>
      </link>
    </model>

    <!-- Omega1 marker -->
    <!-- <model name="omega1_marker">
      <static>true</static>
      <pose>-2 2 0 0 0 0</pose>
      <link name="link">
        <visual name="marker">
          <geometry>
            <sphere>
              <radius>0.05</radius>
            </sphere>
          </geometry>
          <material>
            <ambient>1 0 1 1</ambient>
            <diffuse>1 0 1 1</diffuse>
          </material>
        </visual>

        <visual name="x_axis">
          <geometry>
            <cylinder>
              <radius>0.02</radius>
              <length>24</length>
            </cylinder>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Red</name>
            </script>
          </material>
          <pose>12 0 0 0 -1.5708 0</pose>
        </visual>
        <visual name="y_axis">
          <geometry><cylinder><radius>0.02</radius><length>24</length></cylinder></geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Green</name>
            </script>
          </material>
          <pose>0 12 0 -1.5708 0 0</pose>
        </visual>
        <visual name="z_axis">
          <geometry><cylinder><radius>0.02</radius><length>24</length></cylinder></geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Blue</name>
            </script>
          </material>
          <pose>0 0 12 0 0 0</pose>
        </visual>
      </link>
    </model> -->

  </world>
</sdf>